storage/local/storage.go

```diff
@@ -1144,37 +1161,80 @@ func (s *memorySeriesStorage) incNumChunksToPersist(by int) {
 	atomic.AddInt64(&s.numChunksToPersist, int64(by))
 }
 
-// isDegraded returns whether the storage is in "graceful degradation mode",
-// which is the case if the number of chunks waiting for persistence has reached
-// a percentage of maxChunksToPersist that exceeds
-// percentChunksToPersistForDegradation. The method is not goroutine safe (but
-// only ever called from the goroutine dealing with series maintenance).
-// Changes of degradation mode are logged.
-func (s *memorySeriesStorage) isDegraded() bool {
-	nowDegraded := s.getNumChunksToPersist() > s.maxChunksToPersist*percentChunksToPersistForDegradation/100
-	if s.degraded && !nowDegraded {
-		log.Warn("Storage has left graceful degradation mode. Things are back to normal.")
-	} else if !s.degraded && nowDegraded {
+// inRushedMode returns whether the storage is in "rushed mode", which is the
+// case if there are too many chunks waiting for persistence or there are too
+// many chunks in memory. The method is not goroutine safe (but only ever called
+// from the goroutine dealing with series maintenance).  Changes of degradation
+// mode are logged.
+func (s *memorySeriesStorage) inRushedMode() bool {
```

Comment:
Go still has getters but generally "get" and "set" as prefixes are hardly used. Sure, Go lacks static constraints to enforce read-only – methods names are still suggestive to the behavior.

To me functions that return are bool and could easily be suffixed with a `?` always seem like getters.
I don't mind too much since it's an internal method.

`checkRushedMode` would be clearer to me – but that's highly subjective.